src/Glpi/UI/IllustrationManager.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\UI;
+
+final class IllustrationManager
+{
+    public const DEFAULT_ICON = "request-service.svg";
+
+    /**
+     * @param int $size Height and width (px)
+     */
+    public function render(string $filename, int $size = 100): string
+    {
+        $svg_content = $this->getSvgContent($filename);
+        $svg_content = $this->replaceColorsByVariables($svg_content);
+        $svg_content = $this->setSize($svg_content, $size);
```

Comment:
This method is not really a getter.
```suggestion
        $svg_content = $this->adjustSize($svg_content, $size);
```